astropy/io/fits/file.py

```diff
@@ -41,7 +41,7 @@
     'copyonwrite': 'rb',
     'update': 'rb+',
     'append': 'ab+',
-    'ostream': 'wb',
+    'ostream': 'wb+',
```

Comment:
That's a bit annoying, since when mapping file mode to FITS mode, 'wb+' maps to 'update'. And until now 'ostream' is writeonly, and it actually means that the `.writeonly` attribute is True. I don't have a better suggestion though, and it seems fine, but this does not help with the clarity of this file/FITS mode mapping (maybe at least add a comment to explain this ?).

Reply:
I've added a comment - if it's not quite what you meant, could you make a suggested edit?